scratch/evaluate_scheduler_qd_dense_scenario.cc

```diff
@@ -0,0 +1,825 @@
+typedef CommunicationPairList::const_iterator CommunicationPairListCI;
+
+/** Simulation Arguments **/
+string applicationType = "onoff";             /* Type of the Tx application */
+string socketType = "ns3::UdpSocketFactory";  /* Socket Type (TCP/UDP) */
+string schedulerType = "ns3::CbapOnlyDmgWifiScheduler";   /* The type of scheduler to be used */
+string phyMode = "DMG_MCS12";                 /* The MCS to be used at the Physical Layer. */
+uint32_t packetSize = 1448;                   /* Application payload size [bytes]. */
+string tcpVariant = "NewReno";                /* TCP Variant Type. */
+uint32_t maxPackets = 0;                      /* Maximum Number of Packets */
+uint32_t msduAggregationSize = 7935;          /* The maximum aggregation size for A-MSDU [bytes]. */
+uint32_t mpduAggregationSize = 262143;        /* The maximum aggregation size for A-MPDU [bytes]. */
+double simulationTime = 10;                   /* Simulation time [s]. */
+bool csv = false;                             /* Enable CSV output. */
+uint8_t allocationId = 1;                     /* The allocation ID of the DMG Tspec element to create */
+
+/* NUmber of SP allocations */
+uint16_t numSPs = 6;
+
+/*** Beamforming CBAP ***/
+uint16_t biThreshold = 10;                    /* BI Threshold to trigger TxSS TXOP. */
+map<Mac48Address, uint16_t> biCounter;   /* Number of beacon intervals that have passed. */
+
+/**  Applications **/
+CommunicationPairList communicationPairList;  /* List of communicating devices. */
+
+/* MAC layer Statistics */
+typedef map<Mac48Address, uint64_t> PacketCountMap;
+PacketCountMap macTxDataFailed;
+PacketCountMap macTxDataOk;
+PacketCountMap macRxDataOk;
+
+Ptr<DmgApWifiMac> apWifiMac;
+
+template <typename T>
+string to_string_with_precision (const T a_value, const int n = 6)
+{
+  ostringstream out;
+  out.precision (n);
+  out << fixed << a_value;
+  return out.str ();
+}
+
+vector<string>
+SplitString (const string &str, char delimiter)
+{
+  stringstream ss (str);
+  string token;
+  vector<string> container;
+
+  while (getline (ss, token, delimiter))
+    {
+      container.push_back (token);
+    }
+  return container;
+}
+
+void
+EnableMyTraces (vector<string> &logComponents, Time tLogStart, Time tLogEnd)
+{
+  for (size_t i = 0; i < logComponents.size (); ++i)
+    {
+      const char* component = logComponents.at (i).c_str ();
+      if (strlen (component) > 0)
+        {
+          NS_LOG_UNCOND ("Logging component " << component);
+          Simulator::Schedule (tLogStart, &LogComponentEnable, component, LOG_LEVEL_ALL);
+          Simulator::Schedule (tLogEnd, &LogComponentDisable, component, LOG_LEVEL_ALL);
+        }
+    }
+}
+
+double
+CalculateSingleStreamThroughput (Ptr<PacketSink> sink, uint64_t &lastTotalRx, double &averageThroughput)
+{
+  double thr = (sink->GetTotalRx () - lastTotalRx) * (double) 8 / 1e5;     /* Convert Application RX Packets to Mbits. */
```

Comment:
It should be because `(sink->GetTotalRx () - lastTotalRx) *  8.0` is bits/100ms. Then times 10 is bits/s and divided by 10^6 is Mbits/s.

Reply:
oh jesus, please fix this mess :D
it can easily lead to stupid bugs and it is not clear at all looking at the code.
I see that the 100ms period is hard coded. You could set it as a global variable and then use it to compute the `thr` with the appropriate EXPLICIT transformations